README.md

```diff
@@ -145,16 +174,17 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
-
+run with:
+python3 state.py
```

Comment:
# 8

What you have written is a very nice recursive descent parser. Unfortunately I asked for one based on a state transition table. Something like this:

~~~ py
import unittest
from functools import reduce
ERROR  = "ERROR"
OK     = "OK"

states = {
  0:  { "the":    1 },
  1:  { "lazy":   2, "smelly":  2, "dog": 3, "cat": 3 },
  2:  { "lazy":   2, "smelly":  2, "dog": 3, "cat": 3 },
  3:  { "ate":    4, "ran":     4 },
  4:  { "slowly": 5, "noisily": 5, "EOI": OK },
  5:  { "EOI":    OK }
}

def getNextState(currentState, nextWord):
    return states.get(currentState,{}).get(nextWord, ERROR)

def stateMachine(sentence):
    return reduce(getNextState, sentence, 0) == OK
~~~